enterprise/api/v1/views.py

```diff
@@ -64,6 +64,71 @@ class EnterpriseCustomerViewSet(EnterpriseReadOnlyModelViewSet):
     filter_fields = FIELDS
     ordering_fields = FIELDS
 
+    @detail_route(
+        permission_classes=[permissions.IsAuthenticated, IsServiceUserOrReadOnly],
+        authentication_classes=[JwtAuthentication, BearerAuthentication, SessionAuthentication],
+        throttle_classes=[ServiceUserThrottle],
+    )  # pylint: disable=invalid-name,unused-argument
+    def courses(self, request, pk=None):
+        """
+        Retrieve the list of courses contained within the catalog linked to this enterprise.
+
+        Only courses with active course runs are returned. A course run is considered active if it is currently
+        open for enrollment, or will open in the future.
+        ---
+        serializer: serializers.CourseSerializerExcludingClosedRuns
+        """
+        enterprise_customer = self.get_object()
+
+        # Make sure there is a catalog associated with the enterprise
+        if not enterprise_customer.catalog:
+            error_message = (
+                "No catalog is associated with Enterprise {enterprise_name} from endpoint '{path}'.".format(
+                    enterprise_name=enterprise_customer.name,
+                    path=request.get_full_path()
+                )
+            )
+            logger.error(error_message)
+            raise NotFound("The resource you are looking for does not exist: " + error_message)
```

Comment:
would it be better to just use the error message instead of just prefixing it with this message? the error message string does get passed and echoed back to the caller.